### PR TITLE
JAVA-618: Randomize contact points list to prevent hotspots.

### PR DESCRIPTION
**Pull request by adutra**
This commit introduces the following enhancements:
- Randomize Cluster's initial contact points list;
- Randomize initial host index in DCAwareRoundRobinPolicy;
- Randomize returned rows from system.peers table in ControlConnection.

### CODE REVIEW

**Comment by tolbertam** [comment]
This doesn't seem to truly randomize host used for the ControlConnection because ControlConnection.connect() uses cluster.metadata.allHosts() which uses Metadata.hosts.values().  Since Metadata.hosts() is a ConcurrentHashMap, the random ordering from before is discarded.  Adding a test that demonstrates this shortly.

**Comment by adutra** [comment]
@tolbertam very good catch, thank you! I transferred the shuffling operation to the ControlConnection itself. It seems that it's not necessary to shuffle the hosts list at Cluster level, as the load balancing policies already randomize the received list (at least the round-robin ones), so I removed that too.

**Comment by tolbertam** [comment]
Changes look good to me, :+1:

File: driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java [hunk 1]
```diff
@@ -74,7 +74,12 @@ public void connect() throws UnsupportedProtocolVersionException {
             return;
 
         // NB: at this stage, allHosts() only contains the initial contact points
-        setNewConnection(reconnectInternal(cluster.metadata.allHosts().iterator(), true));
+        List<Host> hosts = new ArrayList<Host>(cluster.metadata.allHosts());
+        // shuffle contact points to spread load evenly across hosts on the very first connection;
```
Comment (olim7t) [inline driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java]: "spread load" is maybe a bit deceptive because the driver is not under any kind of load at init, I suggest rewording to "shuffle so that multiple clients with the same contact points don't all pick the same control host".

File: driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java [hunk 2]
```diff
@@ -558,6 +563,10 @@ private static void refreshNodeListAndTokenMap(Connection connection, Cluster.Ma
             allTokens.add(row.getSet("tokens", String.class));
         }
 
+        // shuffle the list of found hosts to
+        // spread the load evenly
+        Collections.shuffle(foundHosts);
```
Comment (olim7t) [inline driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java]: I don't think shuffling makes any difference here: the elements of `foundHosts` are eventually added to `Cluster.metadata.hosts`, which is a concurrent hash map.
We'll have to rely on load balancing policies to properly randomize their query plans.
